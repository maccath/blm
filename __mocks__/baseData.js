const baseData = {
  site: {
    siteMetadata: {
      author: "Site Author",
      description: "Site description",
      title: "Site Title",
    },
  },
  placeholderImage: {
    childImageSharp: {
      fluid: {
        src: "icon.png",
        srcSet: "icon.png 50w, icon.png 100w",
        sizes: "(max-width: 200px) 100vw, 200px",
        base64: "data:image/png;base64,iVBORw0KG",
        aspectRatio: 1,
      },
    },
  },
}

export default baseData
