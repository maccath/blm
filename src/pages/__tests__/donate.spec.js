import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import "jest-styled-components"
import baseData from "../../../__mocks__/baseData"

import Donate from "../donate"
import Layout from "../../components/layout"

beforeEach(() => {
  useStaticQuery.mockImplementation(() => baseData)
})

describe("Donate page", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Layout>
          <Donate />
        </Layout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
