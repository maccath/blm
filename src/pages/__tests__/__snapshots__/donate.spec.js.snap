// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donate page renders correctly 1`] = `
<div
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 500ms ease-in-out",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0 1.0875rem 1.45rem",
      }
    }
  >
    <main>
      <h1>
        Donate
      </h1>
      <p>
        There are many ways to use our financial privilege to combat the current
injustices faced by Black people in America, as well as dismantling white
supremacy and race inequality at home.
      </p>
      <p>
        <strong>
          <a
            href="https://www.joincampaignzero.org/"
          >
            Campaign Zero
          </a>
        </strong>
         - to limit police
interventions, improve community interactions, and ensure accountability;
leading to fewer deaths at the hands of US police.
      </p>
      <p>
        <strong>
          <a
            href="https://secure.actblue.com/donate/bail_funds_george_floyd"
          >
            Bail Funds
          </a>
        </strong>
         -
to secure the release of nonviolent protestors placed in jail pre-trial,
to get them fed, and to a place of safety during the US protests.
      </p>
      <p>
        <strong>
          <a
            href="https://blacklivesmatter.com"
          >
            Black Lives Matter
          </a>
        </strong>
         and

        <strong>
          <a
            href="https://www.gofundme.com/f/ukblm-fund"
          >
            UK BLM
          </a>
        </strong>
         - to fight for Freedom,
Liberation and Justice.
      </p>
      <p>
        <strong>
          <a
            href="https://www.runnymedetrust.org"
          >
            Runnymede
          </a>
        </strong>
         - to fund an independent think
tank challenging race inequality in Britain.
      </p>
      <p>
        <strong>
          <a
            href="https://www.gofundme.com/f/exist-loudly-fund-to-support-queer-black-yp"
          >
            Exist Loudly
          </a>
        </strong>
         -
to support queer Black young people.
      </p>
      <p>
        <strong>
          <a
            href="https://www.gofundme.com/f/black-minds-matter-uk"
          >
            Black Minds Matter
          </a>
        </strong>
         -
to help Black people access appropriate mental health services in the UK.
      </p>
      <p>
        <strong>
          <a
            href="https://www.ukblackpride.org.uk"
          >
            UK Black Pride
          </a>
        </strong>
         - to celebrate African,
Asian, Middle Eastern, Latin American and Caribbean-heritage LGBTQ people.
      </p>
      <p>
        <strong>
          <a
            href="https://www.theblackcurriculum.com"
          >
            The Black Curriculum
          </a>
        </strong>
         - to address
the lack of Black British history in the UK Curriculum.
      </p>
      <div
        className="mt-6 flex space-x-4"
      >
        <a
          class="flex-initial rounded-lg bg-white text-purple p-3 no-underline"
          href="/learn"
        >
          I care and want to learn more.
        </a>
        <a
          class="flex-initial rounded-lg bg-white text-purple p-3 no-underline"
          href="/amplify"
        >
          I'd like to show solidarity.
        </a>
      </div>
    </main>
    <footer
      className="flex justify-center space-x-5 css-1bxtzlt"
    >
      <div>
        <a
          href="https://thenounproject.com/cesarreynoso562/"
        >
          Fist icon by Cesar Reynoso
        </a>
      </div>
      <div>
        <a
          href="https://github.com/maccath/blm"
        >
          <svg
            className="inline mr-2"
            fill="currentColor"
            height={15}
            viewBox="0 0 24 24"
            width={15}
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              GitHub
            </title>
            <path
              d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
            />
          </svg>
          Contribute on GitHub
        </a>
      </div>
    </footer>
  </div>
</div>
`;
